### PR TITLE
feat: optimize ledgerv1

### PR DESCRIPTION
**Pull request by gfyrag**

### CODE REVIEW

File: pkg/ledger/ledger.go [hunk 1]
```diff
@@ -54,9 +54,9 @@ func NewLedger(store Store, monitor Monitor, cache *ristretto.Cache, options ...
 }
 
 func (l *Ledger) Close(ctx context.Context) error {
-	if err := l.store.Close(ctx); err != nil {
-		return errors.Wrap(err, "closing store")
-	}
+	//if err := l.store.Close(ctx); err != nil {
```
Comment (paul-nicolas) [inline pkg/ledger/ledger.go:57]: Should we keep the store close ?

File: pkg/storage/sqlstorage/store_ledger.go [hunk 2]
```diff
@@ -59,11 +67,14 @@ func (s *Store) Close(ctx context.Context) error {
 
 func NewStore(schema Schema, executorProvider func(ctx context.Context) (executor, error),
 	onClose, onDelete func(ctx context.Context) error) *Store {
+
 	return &Store{
 		executorProvider: executorProvider,
 		schema:           schema,
 		onClose:          onClose,
 		onDelete:         onDelete,
+		bloom:            bloom.NewWithEstimates(1000000, 0.01), // TODO: Configure
```
Comment (paul-nicolas) [inline pkg/storage/sqlstorage/store_ledger.go]: Do you want to make that configurable ?

File: pkg/storage/sqlstorage/accounts.go [hunk 3]
```diff
@@ -265,11 +271,20 @@ func (s *Store) ensureAccountExists(ctx context.Context, account string) error {
...
+
 	_, err = executor.ExecContext(ctx, sqlq, args...)
 	return s.error(err)
 }
 
 func (s *Store) UpdateAccountMetadata(ctx context.Context, address string, metadata core.Metadata, at time.Time) error {
+
+	entry, ok := s.cache.Get(address)
+	if ok {
+		account := entry.(*core.AccountWithVolumes)
```
Comment (paul-nicolas) [inline pkg/storage/sqlstorage/accounts.go:284]: Is a lock needed here when modifying the cache pointer ? [](edited-with-Axolo)

File: pkg/storage/sqlstorage/volumes.go [hunk 4]
```diff
@@ -10,8 +10,18 @@ import (
 )
 
 func (s *Store) updateVolumes(ctx context.Context, volumes core.AccountsAssetsVolumes) error {
-	for account, accountVolumes := range volumes {
-		accountBy, err := json.Marshal(strings.Split(account, ":"))
+	for address, accountVolumes := range volumes {
+
+		entry, ok := s.cache.Get(address)
+		if ok {
+			account := entry.(*core.AccountWithVolumes)
```
Comment (paul-nicolas) [inline pkg/storage/sqlstorage/volumes.go]: Should we update the cache after inserting in the database ?

File: pkg/ledger/ledger.go (hunk 1 above)
Comment (gfyrag) [reply pkg/ledger/ledger.go:57]: Oh, i've forgot to close! thks

**Review by paul-nicolas — APPROVED** [review]

**Review by flemzord — APPROVED** [review]

**Pull request merged**